src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,138 @@
+package nearsoft.academy.bigdata.recommendation;
+
+import org.apache.mahout.cf.taste.common.TasteException;
+import org.apache.mahout.cf.taste.impl.model.file.FileDataModel;
+import org.apache.mahout.cf.taste.impl.neighborhood.ThresholdUserNeighborhood;
+import org.apache.mahout.cf.taste.impl.recommender.GenericUserBasedRecommender;
+import org.apache.mahout.cf.taste.impl.similarity.PearsonCorrelationSimilarity;
+import org.apache.mahout.cf.taste.model.DataModel;
+import org.apache.mahout.cf.taste.neighborhood.UserNeighborhood;
+import org.apache.mahout.cf.taste.recommender.RecommendedItem;
+import org.apache.mahout.cf.taste.recommender.UserBasedRecommender;
+import org.apache.mahout.cf.taste.similarity.UserSimilarity;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.*;
+//import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.zip.GZIPInputStream;
+
+public class MovieRecommender{
+
+    int reviews = 0, userNum = 1, productNum = 1;
+    HashMap<String, Integer> users = new HashMap<String, Integer>();
+    HashMap<String, Integer> products = new HashMap<String, Integer>();
+
+
+public MovieRecommender(String file){
+    List<String> usersList = new ArrayList<String>();
+    List<String> productsList = new ArrayList<String>();
+    List<String> scoresList = new ArrayList<String>();
+
+    try{
+        InputStream fileStream = new FileInputStream(file);
+        InputStream gzipStream = new GZIPInputStream(fileStream);
+        Reader decoder = new InputStreamReader(gzipStream);
+        BufferedReader buffered = new BufferedReader(decoder);
+
+        PrintWriter pw = new PrintWriter(new File("movies.csv"));
+
+        String line;
+        StringBuilder sb = new StringBuilder();
+
+        while ((line = buffered.readLine()) != null){
+            if(line.contains("review/userId:")){
+                String[] parts = line.split("review/userId: ");
+                if(!users.containsKey(parts[1])) {
+                    users.put(parts[1], userNum);
+                    usersList.add("" + userNum);
```

Comment:
For this concatenation, I would recommend to use the function `Stirng.valueOf(number)` instead :)